---
title: "Spotify Blog Post HW2"
author: "Emily Peters"
categories: [python]
---

In [46]:
import pandas as pd
spotify = pd.read_csv('https://bcdanl.github.io/data/spotify_all.csv')
spotify

,pid,playlist_name,pos,artist_name,track_name,duration_ms,album_name
0,0,Throwbacks,0,Missy Elliott,Lose Control (feat. Ciara & Fat Man Scoop),226863,The Cookbook
1,0,Throwbacks,1,Britney Spears,Toxic,198800,In The Zone
2,0,Throwbacks,2,Beyoncé,Crazy In Love,235933,Dangerously In Love (Alben für die Ewigkeit)
3,0,Throwbacks,3,Justin Timberlake,Rock Your Body,267266,Justified
4,0,Throwbacks,4,Shaggy,It Wasn't Me,227600,Hot Shot
...,...,...,...,...,...,...,...
198000,999998,✝️,6,Chris Tomlin,Waterfall,209573,Love Ran Red
198001,999998,✝️,7,Chris Tomlin,The Roar,220106,Love Ran Red
198002,999998,✝️,8,Crowder,Lift Your Head Weary Sinner (Chains),224666,Neon Steeple
198003,999998,✝️,9,Chris Tomlin,We Fall Down,280960,How Great Is Our God: The Essential Collection


Cody Johnson has been one of my favorite artists to listen to recently, so his name caught my eye while looking through the spotify data frame.

In [47]:
artist= spotify.query("artist_name=='Cody Johnson'")
artist

,pid,playlist_name,pos,artist_name,track_name,duration_ms,album_name
502,9,old country,9,Cody Johnson,Diamond In My Pocket,196906,A Different Day
30193,456,Country,56,Cody Johnson,Dance Her Home,174680,Cowboy Like Me
30194,456,Country,57,Cody Johnson,Me and My Kind,156200,Cowboy Like Me
30195,456,Country,58,Cody Johnson,Texas Kind Of Way,249573,Six Strings One Dream
30197,456,Country,60,Cody Johnson,Diamond In My Pocket,196906,A Different Day
30198,456,Country,61,Cody Johnson,Lucky,186813,Cowboy Like Me
30235,456,Country,98,Cody Johnson,18 Wheels,180040,A Different Day
30236,456,Country,99,Cody Johnson,Jesus Ain't Watching,219120,A Different Day
30237,456,Country,100,Cody Johnson,Ride With Me,215600,A Different Day
30238,456,Country,101,Cody Johnson,Chain Drinkin',189760,Gotta Be Me


In [ ]:
artist['playlist_name'].nunique()

12

Cody Johnson's music appeared in 12 different playlists.  

In [48]:
tracks=(
    artist
    .drop_duplicates(subset=['track_name'])
    ['track_name'].nunique()
)
tracks

27

In [49]:
artist['album_name'].nunique()

5

27 different songs were featured, and they were from 5 different albums.

In [ ]:
artist['duration_min']=artist['duration_ms']/60000
artist['duration_min'].mean()

On average, Cody Johnson's songs are 3.53 minutes long.

In [51]:
total_ms= artist['duration_ms'].sum()
total_ms

10368578

In [ ]:
total_ms/60000

172.80963333333332

Playing all of Cody Johnson's music from spotify will take 172.81 minutes.